src/net/peacefulcraft/sco/swordskills/utilities/Modifier.java

```diff
@@ -0,0 +1,220 @@
+            if(s == null) { return null; }
+            try{
+                return valueOf(s.toUpperCase());
+            } catch(IllegalArgumentException e) {
+                try{
+                    EntityType et = EntityType.valueOf(s.toUpperCase());
+                    if(et == null) { return null; }
+                    return CUSTOM;
+                } catch(Exception ex) {
+                    return null;
+                }
+            }
+        }
+    }
+
+    /**The Type of mob it is effective against */
+    ModifierType type;
+    /**The amount to modify the damage*/
+    double multiplier;
+    /**The type of calculation we do. */
+    MultiplierType mType;
+    /**True if strength, false if weakness */
+    Boolean strength;
+
+    /**
+     * Constructs baseline modifier.
+     */
+    public Modifier(ModifierType type, double multiplier, MultiplierType mType, boolean strength) {
+        this.type = type;
+        this.multiplier = multiplier;
+        this.mType = mType;
+        this.strength = strength;
+    }
+
+    /**
+     * Constructs modifier from string.
+     * String should be formatted: ZOMBIE-1.2-MULTIPLY
+     * @param strength True if strength, False if weakness
+     */
+    public Modifier(String s, boolean strength) {
+        String[] split = s.split("-");
+        this.type = ModifierType.get(split[0]);
+        this.multiplier = Double.valueOf(split[1]);
+        this.mType = MultiplierType.get(split[2]);
+        this.strength = strength;
+    }
+
+    /**
+     * Calculations done specifically for EntityDamageEvents only
+     * Example: Drowning, explosion, fire, magic(potions), lightning, etc.
+     * Handles creeper explosions.
+     * Only use for incoming damage.
+     */
+    public double calculate(EntityDamageEvent e) {
+        double d = e.getDamage();
+        Entity ent = e.getEntity();
+
+        //Event is caused by entity
+        if(e.getCause().equals(DamageCause.ENTITY_ATTACK) && e instanceof EntityDamageByEntityEvent) {
+            d = calculateByEntity((EntityDamageByEntityEvent)e);
+        }
+        //Victim is mythic mob
+        if(SwordCraftOnline.getPluginInstance().getMobManager().getMobRegistry().containsKey(ent.getUniqueId())) {
+            //Checking if modifier type matches damage type
+            if(this.type.toString().equalsIgnoreCase(e.getCause().toString())) {
+                d = internalCalc(e);
+            }
+        }
+        //Victim is player
+        if(ent instanceof Player && GameManager.findSCOPlayer((Player)ent) != null) {
+            if(this.type.toString().equalsIgnoreCase(e.getCause().toString())) {
+                d = internalCalc(e);
+            }
+        }
+        return d;
+    }
+
+    /**
+     * Calculations done specifically for EntityDamageEntity only
+     */
+    private double calculateByEntity(EntityDamageByEntityEvent e) {
+        Entity ent;
+        double d = e.getDamage();
```

Comment:
I might be reading into this too much, but do we know that `.getDamage()` returns the damage done after armor and potion effects have been calculated? I assume we'd want those modifiers in our calculations and the [docs](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/entity/EntityDamageEvent.html) say `.getDamage()` returns the raw value. 
There is a `.getFinalDamage()` method that I assume does that if `getDamage()` does not. If this has been tested and `.getDamage()` definitely returns the values we want then disregard this. Just wasn't sure if we're using the right value or if I am reading too far into the docs.